library/cartridge_instance.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/python
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.helpers import ModuleRes, CartridgeException
+from ansible.module_utils.helpers import get_control_console
+
+import os
+
+
+argument_spec = {
+    'control_sock': {'required': True, 'type': 'str'},
+    'config': {'required': True, 'type': 'dict'},
+    'cartridge_defaults': {'required': True, 'type': 'dict'},
+}
+
+
+def manage_instance(params):
+    config = params['config']
+    cartridge_defaults = params['cartridge_defaults']
+    control_sock = params['control_sock']
+
+    # Check if memtx_memory parameter is specified
+    if 'memtx_memory' not in config and 'memtx_memory' not in cartridge_defaults:
+        return ModuleRes(success=True, changed=False)
+
+    new_memtx_memory = None
+    if 'memtx_memory' in config:
+        new_memtx_memory = config['memtx_memory']
+    else:
+        new_memtx_memory = cartridge_defaults['memtx_memory']
+
+    # Check if instance is started
+    if not os.path.exists(control_sock):
+        return ModuleRes(success=True, changed=False)
+
+    control_console = get_control_console(control_sock)
+
+    # Get current memtx memory
+    current_memtx_memory = control_console.eval('''
+        return type(box.cfg) ~= 'function' and box.cfg.memtx_memory or require('json').NULL
+    ''')
+
+    if new_memtx_memory <= current_memtx_memory:
+        return ModuleRes(success=True, changed=False)
+
+    # try to increase memtx_memory
+    increased = control_console.eval('''
+        local ok, err = pcall(function()
+            box.cfg {{ memtx_memory = {} }}
+        end)
+        if not ok then
+            if tostring(err):find("cannot decrease memory size at runtime") == nil then
```

Comment:
It's not a solution. You need to find another way. Try to google about `box.error` submodule: you need to get last box error.